### PR TITLE
Tutor impersonates a minor

### LINKED ISSUE #2: [GDPR15] Manage a minor's user account
Reference GDPR15

**Is your feature request related to a problem? Please describe.**
As the guardian of a minor's account, I need a way to audit the actions she has taken. 

**Describe the solution you'd like**
I would like to be able to manage this account, i.e. log in with this user's account for a certain period of time.

**Describe alternatives you've considered**
To be able to have a record of the minor's account activity, without the need to access his or her account.

**Does this issue could impact on users private data?**
Yes, as an adult is accessing a minor's account and this data is particularly sensitive.

**Acceptance criteria**
- [x] As a guardian with a minor account associated, I can log in with the minor account.
- [x] As a guardian with a minor account associated logged in with the minor account, in a specific period of time the minor session will be automatically closed.
- [x] Allow to impersonate only if a variable is checked in the /system panel

### PR DESCRIPTION
**Pull request by antopalidi**
In this PR, the tutor can impersonate a minor.

- Impersonation of a minor is allowed if the minor has logged in at least once.
- The tutor can end the session before the end of time
- After the time expires, the session automatically ends

screen
screen

screen

### CODE REVIEW

**Review by microstudi — DISMISSED** [review]
good!

**Comment by microstudi** [comment]
@fblupi this should be ready to review

File: app/commands/decidim/kids/close_session_managed_minor.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Kids
+    # A command with all the business logic to close a current impersonation session.
+    class CloseSessionManagedMinor < Decidim::Command
+      # Public: Initializes the command.
+      #
+      # current_user       - The user impersonating a minor
+      # minor_user         - The user to impersonate
```
Comment (fblupi) [inline app/commands/decidim/kids/close_session_managed_minor.rb]: There is a difference between the attribute in the comment and the attribute used in the method: `minor_user` vs `real_user`

File: app/commands/decidim/kids/impersonate_minor.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+module Decidim
+  module Kids
+    # A command with all the business logic to impersonate a minor.
+    class ImpersonateMinor < Decidim::Command
+      # Public: Initializes the command.
+      #
+      # current_user       - The user impersonating a minor
+      # minor_user         - The user to impersonate
+      def initialize(minor_user, current_user, form)
```
Comment (fblupi) [inline app/commands/decidim/kids/impersonate_minor.rb:12]: `form` attribute should be documented above

**Review by fblupi — CHANGES_REQUESTED** [review]

**Review by microstudi — DISMISSED** [review]

**Review by fblupi — APPROVED** [review]

**Pull request merged**